docs/blocks-scaffolding.md

```diff
@@ -0,0 +1,239 @@
+- [ahmadawais/Gutenberg-Boilerplate](https://github.com/ahmadawais/Gutenberg-Boilerplate) - an inline documented starter WordPress plugin for the new Gutenberg editor
+
+## WP-CLI
+
+Another way of making developer's life easier is to use [WP-CLI](http://wp-cli.org/), which provides a command-line interface for many actions you might perform on the WordPress instance. One of the commands generates all the code required to register a Gutenberg block for a plugin or theme. 
+
+### Installing
+
+Before installing `WP-CLI`, please make sure your environment meets the minimum requirements:
+
+* UNIX-like environment (OS X, Linux, FreeBSD, Cygwin); limited support in Windows environment
+* PHP 5.3.29 or later
+* WordPress 3.7 or later
+
+Once you’ve verified requirements, you should follow the [installation instructions](http://wp-cli.org/#installing). Downloading the Phar file is the recommended installation method for most users. Should you need, see also the documentation on [alternative installation methods](https://make.wordpress.org/cli/handbook/installing/).
+
+_Important_: To use scaffolding command for blocks you temporary need (until v1.5.0 is released) to run `wp cli update --nightly` to use the latest nightly build of `WP-CLI`. The nightly build is more or less stable enough for you to use in your development environment, and always includes the latest and greatest `WP-CLI` features.
+
+### Using `wp scaffold block`
+
+The following command generates PHP, JS and CSS code for registering a Gutenberg block.
+
+```bash
+wp scaffold block <slug> [--title=<title>] [--dashicon=<dashicon>] [--category=<category>] [--theme] [--plugin=<plugin>] [--force]
+```
+
+Please refer to the [command documentation](https://github.com/wp-cli/scaffold-command#wp-scaffold-block) to learn more about the available options for the block.
+
+When you scaffold a block you must provide at least a `slug` name and either the `theme` or `plugin` name. In most cases, we recommended pairing blocks with plugins rather than themes, because only using plugin ensures that all blocks will still work when your theme changes.
+
+### Examples
+
+Here are some examples using `wp scaffold block` in action.
+
+#### Create a block inside the plugin
+
+The following command generates a `movie` block with the `My movie block` title for the existing plugin named `movies`:
+
+```bash
+$ wp scaffold block movie --title="My movie block" --plugin=movies
+Success: Created block 'My movie block'.
+```
+
+This will generate 4 files inside the `movies` plugin directory. All files contain inline documentation that will help to apply any further customizations to the block. It's worth mentioning that it is currently possible to scaffold only blocks containing static content and JavaScript code is written using ECMAScript 5 (ES5) standard which works with all modern browsers.
+
+`movies/blocks/movie.php` - you will have to manually include this file in your main plugin file:
```

Comment:
> you will have to manually include this file in your main plugin file

@schlessera and @danielbachhuber, any hints how to automate it? I'm happy to fix it :)

Reply:
You cannot really automate any of this type of stuff, as there is neither interfaces nor conventions for WordPress plugins. It is basically: "If you tag a file in that directory with this header tag, it will get eval'ed!". For more complex plugins, this would be included somewhere deep in a class hierarchy, whereas very simple plugins will prefer to copy-paste it into their single file.

Reply:
Thanks for making it clear 👍